sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala

```diff
@@ -413,4 +413,8 @@ object ShimLoader extends Logging {
   def newPlanShims(): PlanShims = ShimLoader.newInstanceOf[PlanShims](
     "com.nvidia.spark.rapids.shims.PlanShimsImpl"
   )
+  
+  def loadGpuColumnVector(): Class[_] = {
```

Comment:
ideally we should have façade Shim trait/interface for GpuCV for type-safe interactions. Then you don't need to use reflection to call the method. 

The signature for the method you need 
```
 def extractHostColumns(table: Table, colType: Array[DataType]): Array[ColumnVector] = {
```

does not introduce shimming issues because only involves relatively stable classes, and only DataType is a Scala class. If this changes we will have to abstract it away further when it comes to it.

you can have something
```
trait GpuColumnVectorShims {
   def extractHostColumns(table: Table, colType: Array[DataType]): Array[ColumnVector] = {
}

class GpuColumnVectorShimsImpl extends GpuColumnVectorShims  {
   def extractHostColumns(table: Table, colType: Array[DataType]): Array[ColumnVector] = 
      GpuColumnVector.extractHostColumns(table, colType)
}
```

And in ShimLoader

```
 def newGpuColumnVectorShims(): GpuColumnVectorShims  = ShimLoader.newInstanceOf[GpuColumnVectorShims ](
    "package.GpuColumnVectorShimsImpl"
  )
```



The user API

ShimLoader.newGpuColumnVectorShims().extractHostColumns(table, colType)

Reply:
I tried your suggestion, but it didn't work. the error showed 

``` console
Caused by: java.lang.ClassNotFoundException: com.nvidia.spark.rapids.GpuColumnVector
	at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
```

Reply:
I'll try to reproduce